pandas/tests/io/json/test_normalize.py

```diff
@@ -460,3 +501,67 @@ def test_nonetype_multiple_levels(self):
             'location.country.state.town.info.y': -33.148521423339844,
             'location.country.state.town.info.z': 27.572303771972656}
         assert result == expected
+
+    @pytest.mark.parametrize("max_level", [None, 0, 1])
+    def test_with_max_level_none(self, max_level):
+        # GH23843: Enhanced JSON normalize
+        data = [{
+            'CreatedBy': {'Name': 'User001'},
+            'Lookup': {'TextField': 'Some text',
+                       'UserField': {'Id': 'ID001', 'Name': 'Name001'}},
+            'Image': {'a': 'b'}
+        }]
+        expected_output = {None: [{'CreatedBy.Name': 'User001',
```

Comment:
Parametrization is looking nice! Though the idiomatic way to do this would be as follows:

```python
@pytest.mark.parametrize("max_level,expected", [
    (None, [{'CreatedBy.Name': 'User001',
      ...),
    (0, [{'CreatedBy': {'Name': 'User001'}.
      ...),
    (1, [{'CreatedBy.Name': {'User001'},
      ...)]
```

i.e. pass in a tuple of value and expectation as part of the parametrization and change the function signature to allow both `max_level` and `expected`

Reply:
Yes will do that. That's a beautiful way to do it.

I don't understand why docs build is failing can you help me with that?

Reply:
Not related to your PR - can you merge master? Might have been resolved in another PR